shared/util/quit-helper.desktop.js

```diff
@@ -27,17 +30,23 @@ function isMainThread () {
 }
 
 function _executeActions (actions: Array<Action>) {
+  // Don't allow us to re-enter this logic if we're already shutting down
+  if (isQuitting) {
```

Comment:
If something fails while quitting and this becomes true, can the user never quit anymore?